### PR TITLE
[8.1] minor py3-related fixes

### PR DESCRIPTION
**Pull request by fstagni**

### CODE REVIEW

**Comment by chrisburr** [comment]
I think these can all target 8.0?

File: src/DIRAC/Core/Utilities/Graphs/Graph.py [hunk 1]
```diff
@@ -275,15 +275,10 @@ def makeGraph(self, data, *args, **kw):
...
-                # TODO: Remove when we moved to python3
-                exec("import %s" % plot_type)
-            except ImportError:
-                print("Trying to use python like import")
-                try:
-                    exec("from . import  %s" % plot_type)
-                except ImportError as x:
-                    print(f"Failed to import graph type {plot_type}: {str(x)}")
-                    return None
+                exec("from . import  %s" % plot_type)
```
Comment (chrisburr) [inline src/DIRAC/Core/Utilities/Graphs/Graph.py]: Can you make this use `importlib` instead?

File: src/DIRAC/DataManagementSystem/Client/FileCatalogClientCLI.py [hunk 2]
```diff
@@ -1,17 +1,12 @@
 #!/usr/bin/env python
 """ File Catalog Client Command Line Interface. """
-# TODO: This should be modernised to use subprocess(32)
-try:
-    import commands
-except ImportError:
-    # Python 3's subprocess module contains a compatibility layer
-    import subprocess as commands
+import subprocess as commands
```
Comment (chrisburr) [inline src/DIRAC/DataManagementSystem/Client/FileCatalogClientCLI.py]: No need to still name it `commands`
```suggestion
import subprocess
```

File: src/DIRAC/Interfaces/API/Dirac.py [hunk 3]
```diff
@@ -51,12 +51,7 @@
 
 COMPONENT_NAME = "DiracAPI"
 
-try:
-    # Python 2: "file" is built-in
-    file_types = file, io.IOBase
-except NameError:
-    # Python 3: "file" fully replaced with IOBase
-    file_types = (io.IOBase,)
+file_types = (io.IOBase,)
```
Comment (chrisburr) [inline src/DIRAC/Interfaces/API/Dirac.py]: Probably best to get rid of this entirely and just use `io.IOBase` where needed
```suggestion
```

**Comment by fstagni** [comment]
I pushed some additional fixes, also touching the "DIRACROOT", "DIRAC.root" etc stuff which is more delicate. I will now go a bit deeper into it so I would keep it here in 8.1.

File: src/DIRAC/DataManagementSystem/Client/FileCatalogClientCLI.py [hunk 4]
```diff
@@ -584,7 +579,7 @@ def registerFile(self, args):
         infoDict["Size"] = int(args[2])
         infoDict["SE"] = args[3]
         if args[4] == "None":
-            _status, guid = commands.getstatusoutput("uuidgen")
+            _status, guid = subprocess.getstatusoutput("uuidgen")
```
Comment (chrisburr) [inline src/DIRAC/DataManagementSystem/Client/FileCatalogClientCLI.py]: You'll also need to import `uuid`:
```suggestion
            guid = str(uuid.uuid4())
```

File: src/DIRAC/Core/Utilities/Graphs/Graph.py [hunk 5]
```diff
@@ -342,8 +337,8 @@ def drawWaterMark(self, imagePath=None):
             ax_wm.axis("off")
             ax_wm.set_frame_on(False)
             ax_wm.set_clip_on(False)
-        except Exception as e:
-            print(e)
+        except Exception:
+            gLogger.exception("Caught expection")
```
Comment (andresailer) [inline src/DIRAC/Core/Utilities/Graphs/Graph.py]:
```suggestion
            gLogger.exception("Caught exception")
```

File: src/DIRAC/Core/Utilities/Graphs/Graph.py [hunk 6]
```diff
@@ -352,4 +347,4 @@ def writeGraph(self, fname, fileFormat="PNG"):
         if fileFormat.lower() == "png":
             self.canvas.print_png(fname)
         else:
-            gLogger.error("File format '%s' is not supported!" % fileFormat)
+            gLogger.error("File format '{%s}' is not supported!")
```
Comment (andresailer) [inline src/DIRAC/Core/Utilities/Graphs/Graph.py]:
```suggestion
            gLogger.error(f"File format '{fileFormat}' is not supported!")
```

File: src/DIRAC/WorkloadManagementSystem/Utilities/Utils.py [hunk 7]
```diff
@@ -31,16 +30,12 @@ def createJobWrapper(
...
-    mkDir("%s/job/Wrapper" % (workingDir))
-
+    mkDir("%s/job/Wrapper" % (os.getcwd()))
     diracRoot = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
 
-    jobWrapperFile = f"{workingDir}/job/Wrapper/Wrapper_{jobID}"
+    jobWrapperFile = f"{os.getcwd()}/job/Wrapper/Wrapper_{jobID}"
     if os.path.exists(jobWrapperFile):
-        log.verbose("Removing existing Job Wrapper for %s" % (jobID))
+        log.verbose("Removing existing Job Wrapper for {jobID}")
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Utilities/Utils.py]:
```suggestion
        log.verbose(f"Removing existing Job Wrapper for {jobID}")
```

File: src/DIRAC/WorkloadManagementSystem/Utilities/Utils.py [hunk 8]
```diff
@@ -31,16 +30,12 @@ def createJobWrapper(
     arguments = {"Job": jobParams, "CE": resourceParams, "Optimizer": optimizerParams}
     log.verbose("Job arguments are: \n %s" % (arguments))
 
-    siteRoot = gConfig.getValue("/LocalSite/Root", os.getcwd())
-    log.debug("SiteRootPythonDir is:\n%s" % siteRoot)
-    workingDir = gConfig.getValue("/LocalSite/WorkingDirectory", siteRoot)
-    mkDir("%s/job/Wrapper" % (workingDir))
-
+    mkDir("%s/job/Wrapper" % (os.getcwd()))
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Utilities/Utils.py]:
```suggestion
    mkDir(os.path.join(os.getcwd(), "job/Wrapper"))
```

**Review by andresailer — APPROVED** [review]

**Pull request merged**